components/layout/display_list_builder.rs

```diff
@@ -1097,7 +1140,10 @@ impl FragmentDisplayListBuilding for Fragment {
         let length = Au::from_f32_px(
             (delta.x.to_f32_px() * 2.0).hypot(delta.y.to_f32_px() * 2.0));
 
-        let stops = convert_gradient_stops(stops, length, style);
+        let mut stops = convert_gradient_stops(stops, length, style);
+        if !repeating {
```

Comment:
Can you elaborate on why not fixing them up if it's a repeating linear gradient, perhaps with a comment?